packages/appstore/src/components/wallet-transfer/wallet-transfer.tsx

```diff
@@ -18,172 +19,269 @@ type TWalletTransferProps = {
 
 const Divider = () => <div className='wallet-transfer__divider' />;
 
-const WalletTransfer = observer(
-    ({ contentScrollHandler, is_wallet_name_visible, setIsWalletNameVisible }: TWalletTransferProps) => {
-        const { ui } = useStore();
-        const { is_mobile } = ui;
+const initial_demo_balance = 10000.0;
+
+const ERROR_CODES = {
+    is_demo: {
+        between_min_max: 'BetweenMinMax',
+        insufficient_fund: 'InsufficientFund',
+    },
+};
+
+const WalletTransfer = observer(({ is_wallet_name_visible, setIsWalletNameVisible }: TWalletTransferProps) => {
+    const { client, ui, traders_hub } = useStore();
+    const { setWalletModalActiveTab } = traders_hub;
+    const { is_switching } = client;
+    const { is_mobile } = ui;
+
+    const { getConfig } = useCurrencyConfig();
+
+    const {
+        active_wallet,
+        is_accounts_loading,
+        from_account,
+        to_account,
+        to_account_list,
+        transfer_accounts,
+        setFromAccount,
+        setToAccount,
+    } = useWalletTransfer();
+
+    useEffect(() => {
+        if (!from_account?.loginid) {
+            setFromAccount(active_wallet);
+        }
+    }, [active_wallet, from_account, setFromAccount]);
+
+    const portal_id = is_mobile ? 'mobile_list_modal_root' : 'modal_root';
+
+    const is_amount_to_input_disabled = !to_account;
+
+    const active_wallet_name = getAccountName({ ...active_wallet });
+
+    const transfer_to_hint = React.useMemo(() => {
+        return to_account?.loginid === active_wallet?.loginid ? (
+            <Localize
+                i18n_default_text='You can only transfers funds from the {{account}} to the linked {{wallet}}.'
+                values={{
+                    account: getAccountName({ ...from_account }),
+                    wallet: active_wallet_name,
+                }}
+            />
+        ) : (
+            ''
+        );
+    }, [active_wallet?.loginid, active_wallet_name, from_account, to_account?.loginid]);
+
+    const [message_list, setMessageList] = React.useState<TMessageItem[]>([]);
+
+    const clearErrorMessages = React.useCallback(
+        () => setMessageList(list => list.filter(el => el.type !== 'error')),
+        []
+    );
 
-        const [from_account, setFromAccount] = React.useState<TAccount>(transfer_accounts.wallets[0]);
-        const [to_account, setToAccount] = React.useState<TAccount>();
+    const validateAmount = (amount: number) => {
+        clearErrorMessages();
 
-        const portal_id = is_mobile ? 'mobile_list_modal_root' : 'modal_root';
+        if (!amount || is_amount_to_input_disabled) return;
 
-        const to_account_list = React.useMemo(() => {
-            // TODO: 'Demo USD Wallet' should be replaced to the current open wallet after connecting to API call
-            if (from_account?.label === 'Demo USD Wallet') {
+        if (active_wallet?.is_demo) {
+            const { is_ok, message } = validNumber(amount.toString(), {
```

Comment:
```suggestion
            const { is_ok, message } = validNumber(amount.toString(), {
```
nit: hmmmm i'm not ok with the naming of `is_ok`